tests/pgautofailover_utils.py

```diff
@@ -551,13 +565,22 @@ def wait_until_state(self, target_state,
 
             events = self.get_events_str()
             pglogs = self.get_postgres_logs()
+            if other_node:
+                other_pglogs = other_node.get_postgres_logs()
+            else:
+                other_pglogs = ""
 
-            if self.pg_autoctl and self.pg_autoctl.run_proc:
+            if self.running():
                 out, err = self.stop_pg_autoctl()
+                if other_node and other_node.running():
+                    other_out, other_err = other_node.stop_pg_autoctl()
+                else:
+                    other_out, other_err = "", ""
+
                 raise Exception("%s failed to reach %s after %d attempts: " \
-                                "\n%s\n%s\n%s\n%s" %
+                                "\n%s\n%s\n%s\n%s\n%s\n%s\n%s" %
```

Comment:
I think we need a label (with empty lines before and after) that describe which node the next output belongs to...